### PR TITLE
Flucca/feat/library assets

### LINKED ISSUE #375: Library assets page
**Design**

- [x] Red lines https://ibm.ent.box.com/file/913340274022

**Dev**

- [x] Sort
- [x] Assets table
  - [x] Color coded type tags
  - [x] Hover states
  - [ ] Empty states
  - [ ] Link out per row

Notes:

1. No pagination needed at the moment. We can add pagination later if needed by our data fetching service.

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #375

No side nav yet, need #464 to be merged in 
Design: https://ibm.ent.box.com/file/913340274022

#### Testing / reviewing

Example url: http://localhost:3000/assets/carbon-charts/library-assets
Test all breakpoints against design, make sure linkout works on asset rows, test empty state

https://user-images.githubusercontent.com/40550942/162341860-e04c5109-e9d0-4cca-97c2-29effe26ddde.mov

https://user-images.githubusercontent.com/40550942/162341865-46765bf2-8c93-4df5-b02e-e66d822530b8.mov

### CODE REVIEW

**Comment by francinelucca** [comment]
Ready for first review.
Missing: clickable row (entire row). needs further discussion, might want to split this into a separate ticket.

**Comment by francinelucca** [comment]
@kingtraceyj added some screen records so you can take a look, happy to jump on a huddle/call as well and review!

**Comment by alisonjoseph** [comment]
@francinelucca quickly looking at this made me realize some updates I need to make to #375 to actually wire it up as a working nav

File: services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/assets/index.js
Comment (alisonjoseph) [inline services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/assets/index.js:2]:
```suggestion
 * Copyright IBM Corp. 2022, 2022
```

**Review by kingtraceyj — CHANGES_REQUESTED** [review]
looing good francine! there are just a few things i want to double check.

**Arrow color:**
I just want to double check that arrow color is `$interactive` i forgot that token in the redlines

This was also in another redline and i did not get it into the library redline but we put 24px padding on xlg breakpoints and 16px at lg, md, and sm break points within the text container. i just wanted to double check to see if that was happening here. 
image

This truncated in the Tags column gets lost on tablet and mobile. It's probably best to not truncate and start scroll early. We could look into adding a hover tooltip but that might not be accessible if a user cannot tab into a table to get that information.
image

also small, it looks like there is a tiny gap between the sort and the table and there should be no space: 
image

**Comment by francinelucca** [comment]
@kingtraceyj addressed comments + added text padding, ready for second review
image

**Review by kingtraceyj — CHANGES_REQUESTED** [review]
hey francine! the changes are looking good. one thing, now that we see it in code, is that the assets title is a little redundant with the nav in the header. we are going to change the title of the page from "Assets" to "Library assets" and take out the "All library assets" title above the table. 

I also wanted to check the width of the dropdown, it should be 4 columns and hang 16px. I can't get my carbon dev tools extension to work so i can't see the grid but it should work like this:
image

**Comment by francinelucca** [comment]
@kingtraceyj updated lg columns! hang lgtm

**Review by kingtraceyj — DISMISSED** [review]
looks good to go!

File: services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/assets/index.js
Comment (alisonjoseph) [inline services/web-app/pages/assets/[host]/[org]/[repo]/[library]/[ref]/assets/index.js:133]:
```suggestion
        <Column sm={4} md={8} lg={{ start: 5, span: 12 }}>
```

I think we can just do this here instead, the library side nav will be controlled from the layout.

**Review by alisonjoseph — COMMENTED** [review]
This looks good to me, other than a super minor comment!

**Review by kingtraceyj — DISMISSED** [review]
looks good!

File: services/web-app/components/catalog-item/catalog-item-meta/catalog-item-meta.js
Comment (jdharvey-ibm) [inline services/web-app/components/catalog-item/catalog-item-meta/catalog-item-meta.js:20]: It seems like this implies that the status key can either come in on the `status` field or on the `status.key` field. Is there any way to adjust that so the key is always present on the same field in the asset?

File: services/web-app/next.config.js
Comment (jdharvey-ibm) [inline services/web-app/next.config.js:92]: Am I right that this would mean we'd have URLs like:
`next.carbondesignsystem.com/assets/some-thing/assets`?

If so, could we maybe chat about that to see if we're heading down a weird path?

Comment (francinelucca) [reply services/web-app/next.config.js:92]: yeah so, a library's url is "/assets/library-slug" and this is a page for a library's assets, so I put it under "/assets/library-slug/assets" , I see how that's weird but it also makes sense to me to put the library's "assets" page under [library-url]/assets 🤔

File: services/web-app/components/catalog-item/catalog-item-meta/catalog-item-meta.js
Comment (francinelucca) [reply services/web-app/components/catalog-item/catalog-item-meta/catalog-item-meta.js:20]: yeah so this is how we have it defined in our schema right now (see comment on: https://carbon-design-system.github.io/carbon-platform/resource-schemas.html#asset-status). We can open up a ticket to standardize at the `getLibraryAssets()` level if desired but that'd fall out of the scope of this PR; I'm just moving this out of `services/web-app/components/catalog-item/catalog-item.js` and into it's own component so I can re-use it

File: services/web-app/next.config.js
Comment (alisonjoseph) [reply services/web-app/next.config.js:92]: The url felt a little weird to me too when I was testing, but agree that unless naming of something changes, it makes sense. Tagging @aubrey-oneal and @kingtraceyj just as an FYI here.

File: services/web-app/components/catalog-item/catalog-item-meta/catalog-item-meta.module.scss
Comment (alisonjoseph) [reply services/web-app/components/catalog-item/catalog-item-meta/catalog-item-meta.module.scss:32]: The svg is just a circle, so not sure that editing the svg would make sense? I think adding a comment that this is to visually align with the text is fine here.

File: services/web-app/next.config.js
Comment (mattrosno) [reply services/web-app/next.config.js:92]: I'll create a separate issue for side nav (and URL) naming to not block this PR.

Comment (kingtraceyj) [reply services/web-app/next.config.js:92]: @mattrosno We are trying to create a distinction between the UI Shell 'Assets' and the library assets. I think it's OK to get specific with what assets you are viewing. You aren't looking at all assets just the specific library ones. Using this logic we could go down the path of putting the word library in front of each nav item, but there's no real need since it's unrelated to the top nav item. We've had multiple people, seperately comment on that it's a little confusing to have both the top level nav item say assets as well as the side nav say assets. i think getting specific on this one category adds clarity.

Comment (alisonjoseph) [reply services/web-app/next.config.js:92]: Updating the first item to be "Library assets" and the url to` /assets/carbon-charts/library-assets` is simple enough, and adds clarity.

Comment (mattrosno) [reply services/web-app/next.config.js:92]: I'm okay with that for this PR. We're looking at side nav in a separate issue to ensure all of the other items, and their order, are right.

Comment (francinelucca) [reply services/web-app/next.config.js]: fixed! new sample url: http://localhost:3000/assets/carbon-charts/library-assets

File: services/web-app/components/catalog-item/catalog-item-meta/catalog-item-meta.js
Comment (francinelucca) [reply services/web-app/components/catalog-item/catalog-item-meta/catalog-item-meta.js:20]: Summarized in #562

**Review by alisonjoseph — DISMISSED** [review]

**Review by alisonjoseph — APPROVED** [review]

File: services/web-app/next.config.js
Comment (jdharvey-ibm) [reply services/web-app/next.config.js]: Some notes to toss in the URL vs nav issue:

Can you refresh my memory about the root-most `/assets`? Are we saying that a Library is an Asset and also Libraries can *have* Assets, some of which might be other Libraries? That's what the current URL scheme implies, but I want to confirm that's the intent.

Is `/assets/button` ever a valid url? Or is it always `/assets/some-library-name/button`?

If it's only ever `/assets/some-library`, then I'd vote for being more explicit in the URL about the type of object you're viewing: `/libraries/carbon-charts`. Or even more explicitly: `/assets/libraries/carbon-charts/assets`.

At the moment, there's a mismatch between the breadcrumb and the URL (see screenshot). Assets, Libraries, and a specific library are referenced as navigation contexts on the page, but the URL is missing the indication of the specific library being nested under the set of all libraries.

If a library is not considered as an asset itself and is only a container of *other* assets, then I'd vote for `/libraries/carbon-charts/assets` as opposed to `/assets/libraries/carbon-charts/assets` so that the left nav becomes the more definitive source for the URL format.

(Marking as unresolved so the thread doesn't get lost until we copy/paste into an issue).

image

File: services/web-app/components/catalog-item/catalog-item-meta/catalog-item-meta.module.scss
Comment (jdharvey-ibm) [reply services/web-app/components/catalog-item/catalog-item-meta/catalog-item-meta.module.scss:33]: Some other food for thought: adjusting the vertical alignment to something like `vertical-align: middle;` should get this to the right spot as well.

File: services/web-app/components/catalog-item/catalog-item-meta/catalog-item-meta.js
Comment (jdharvey-ibm) [reply services/web-app/components/catalog-item/catalog-item-meta/catalog-item-meta.js:20]: Oof, that definitely makes parsing the schema harder, but I see the value in the convenience for the end user to be able to forego the note if all they want to add is the key. Thanks for opening the issue!

**Review by jharvey10 — APPROVED** [review]

**Pull request merged**
